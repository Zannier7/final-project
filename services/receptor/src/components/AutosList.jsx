import React from 'react';

const AutosList = (props) => {
  return (
    <div>
      {
        props.autos.map((autos) => {
          return (
            <h4
              key={autos.id}
              className="box title is-4"
            >{ autos.marca }
            </h4>
          )
        })
      }
    </div>
  )
};

export default AutosList;
