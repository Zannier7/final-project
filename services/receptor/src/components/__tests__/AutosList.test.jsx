import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import AutosList from '../AutosList';

const autos = [
    {
        'color':'blanco',
        'id':1,
        'marca':'bmw',
        'modelo':'ci3',
        'placa':'7u456a',
        'tipo':'deportivo'
    },
    {
        'color':'negro',
        'id':2,
        'marca':'audio',
        'modelo':'uik',
        'placa':'5625as',
        'tipo':'4x4'
    },
    {
        'color':'negro-gris',
        'id':3,
        'marca':'volvo',
        'modelo':'mc7',
        'placa':'45as15',
        'tipo':'suv'
    }
];

test('AutosList renders properly', () => {
  const wrapper = shallow(<AutosList autos={autos}/>);
  const element = wrapper.find('h4');
  expect(element.length).toBe(3);
  expect(element.get(0).props.children).toBe('bmw');
});

test('AutosList renders a snapshot properly', () => {
  const tree = renderer.create(<AutosList autos={autos}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
