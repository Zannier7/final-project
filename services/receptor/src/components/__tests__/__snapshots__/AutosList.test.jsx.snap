// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutosList renders a snapshot properly 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    bmw
  </h4>
  <h4
    className="box title is-4"
  >
    audio
  </h4>
  <h4
    className="box title is-4"
  >
    volvo
  </h4>
</div>
`;
